beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -399,14 +414,27 @@ func setExecution(blk interfaces.SignedBeaconBlock, execution interfaces.Executi
 	}
 
 	// Set the KZG commitments for the block
-	errMessage = "failed to set local kzg commitments"
+	kzgErr := "failed to set local kzg commitments"
 	if isBlinded {
-		errMessage = "failed to set builder kzg commitments"
+		kzgErr = "failed to set builder kzg commitments"
 	}
 	if err := blk.SetBlobKzgCommitments(kzgCommitments); err != nil {
-		return errors.Wrap(err, errMessage)
+		return errors.Wrap(err, kzgErr)
 	}
 
+	// If the block version is below Electra, no further actions are needed
+	if blk.Version() < version.Electra {
+		return nil
+	}
+
+	// Set the execution requests
+	requestsErr := "failed to set local execution requests"
+	if isBlinded {
+		requestsErr = "failed to set builder execution requests"
+	}
+	if err := blk.SetExecutionRequests(requests); err != nil {
```

Comment:
If a builder returns us an invalid execution requests, shouldn't we default to local block?

Reply:
isn't that what we do? it returns an error and then it falls back to 
```
	if err := setBuilderExecution(blk, builderPayload, builderKzgCommitments, executionRequests); err != nil {
				log.WithError(err).Warn("Proposer: failed to set builder payload")
				return local.Bid, local.BlobsBundle, setLocalExecution(blk, local)
```